lib/pos_mempool.go

```diff
@@ -169,6 +174,11 @@ func (mp *PosMempool) Start() error {
 
 	// Create the transaction register, the ledger, and the nonce tracker,
 	mp.txnRegister = NewTransactionRegister(mp.globalParams)
+	mp.feeEstimator = NewPoSFeeEstimator()
```

Comment:
You can set `mp.feeEstimator = NewPoSFeeEstimator()` in the `NewPosMempool` constructor above. This would guarantee that the estimator is always are non-nil once the mempool has been constructed